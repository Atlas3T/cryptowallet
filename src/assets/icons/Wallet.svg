<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 22.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 51 51" style="enable-background:new 0 0 51 51;" xml:space="preserve">
<style type="text/css">
	.st0{fill:none;stroke:#000000;stroke-width:5;stroke-miterlimit:10;}
	.st1{display:none;}
	.st2{display:inline;fill:none;stroke:#000000;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:10;}
	.st3{display:inline;}
</style>
<g id="Wallet">
	<path class="st0" d="M46.1,31.5v5.3c0,2.4-1.9,4.3-4.3,4.3H9.5c-2.4,0-4.3-1.9-4.3-4.3V15.5c0-2.4,1.9-4.3,4.3-4.3h32.3
		c2.4,0,4.3,1.9,4.3,4.3v5.2"/>
	<circle cx="46.1" cy="26.1" r="3.1"/>
</g>
<g id="Payments" class="st1">
	<line class="st2" x1="8" y1="19.8" x2="41.8" y2="19.8"/>
	<polyline class="st2" points="35.8,13.3 42.3,19.8 35.8,26.3 	"/>
	<line class="st2" x1="42.3" y1="32.8" x2="8.5" y2="32.8"/>
	<polyline class="st2" points="14.5,39.3 8,32.8 14.5,26.3 	"/>
</g>
<g id="Exchange" class="st1">
	<path class="st2" d="M15.3,21.7c0-5.6,4.6-10.2,10.2-10.2s10.2,4.6,10.2,10.2"/>
	<polyline class="st2" points="42.2,16.3 35.7,22.8 29.2,16.3 	"/>
	<path class="st2" d="M35.7,30.2c0,5.6-4.6,10.2-10.2,10.2s-10.2-4.6-10.2-10.2"/>
	<polyline class="st2" points="8.8,35.7 15.3,29.2 21.9,35.7 	"/>
</g>
<g id="More" class="st1">
	<circle class="st3" cx="26" cy="15.8" r="7.8"/>
	<circle class="st3" cx="12.3" cy="28.6" r="5"/>
	<circle class="st3" cx="35.3" cy="33.6" r="3.5"/>
</g>
</svg>
