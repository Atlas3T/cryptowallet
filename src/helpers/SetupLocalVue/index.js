import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import VueI18n from 'vue-i18n';
import messages from '@/i18n/';
import axios from 'axios';
import { sync } from 'vuex-router-sync';
import errorHandlerPlugin from '@/boot/ErrorHandler';
import toasterPlugin from '@/boot/Toaster';
import permissionsPlugin from '@/boot/Permissions';
import appInvitationPlugin from '@/boot/AppInvitation';
import smsPlugin from '@/boot/Sms';
import emailPlugin from '@/boot/Email';
import contactsImportPlugin from '@/boot/ContactsImport';
import * as All from 'quasar'

const { Quasar, date } = All;

const components = Object.keys(All).reduce((object, key) => {
  const val = All[key]
  if (val && val.component && val.component.name != null) {
    object[key] = val
  }
  return object
}, {})


const localVue = createLocalVue();

localVue.use(Quasar, { components });
localVue.use(VueI18n);
localVue.use(Vuex);
// localVue.use(VueRouter);
localVue.prototype.$axios = axios;
errorHandlerPlugin({ Vue: localVue });
toasterPlugin({ Vue: localVue });
permissionsPlugin({ Vue: localVue });
appInvitationPlugin({ Vue: localVue });
smsPlugin({ Vue: localVue });
emailPlugin({ Vue: localVue });
contactsImportPlugin({ Vue: localVue });

/**
 * creates a new router and syncs it with the store using vuex-router-sync
 * @param store
 * @returns router
 */
const createRouter = (store = null) => {
  const router = new VueRouter({
    routes: [
      {
        path: '/contact/:id',
      },
    ],
  });

  if (store) { sync(store, router); }
  return router;
};

const i18n = new VueI18n({ messages, locale: 'en-gb' });

export {
  localVue,
  createRouter,
  i18n,
};
