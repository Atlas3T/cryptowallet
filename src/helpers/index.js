import AmountFormatter from './AmountFormatter';
import dateTranslater from './DateTranslater';
import Contacts from './Contacts';

export {
  AmountFormatter,
  dateTranslater,
  Contacts,
};
