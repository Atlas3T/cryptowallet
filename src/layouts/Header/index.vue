<template>
  <section
    v-if="!hideHeader"
    class="header-section"
  >
    <div class="header-back-button-wrapper">
      <q-btn
        :disable="!isBackButtonEnabled"
        icon="arrow_back"
        color="primary"
        size="lg"
        class="icon-btn back-arrow-btn"
        flat
        @click.prevent="goBack"
      />
    </div>

    <div v-if="coinHeading">
      <img
        :src="coinLogo"
        class="coin-logo"
      >
      {{ wallet.displayName }}
    </div>
    <div v-else>
      <div v-if="heading === 'CryptoWallet'">
        <h1 class="header-h1 logo">{{ heading }}</h1>
      </div>
      <div v-else>
        <h1 class="header-h1">{{ heading }}</h1>
      </div>
    </div>

    <div
      v-if="displaySettings"
      class="header-settings-button-wrapper"
    >
      <q-btn
        :label="$t('settings')"
        color="secondary"
        size="sm"
        @click.prevent="goToSettings"
      />
    </div>

    <div
      v-if="displayAddWallet"
      class="header-settings-button-wrapper"
    >
      <q-btn
        icon="add"
        color="primary"
        size="lg"
        class="icon-btn icon-btn-right"
        flat
        @click.prevent="openWalletsModal"
      />
    </div>

    <div
      v-if="displayPriceChart"
      class="header-settings-button-wrapper"
    >
      <q-btn
        icon="timeline"
        color="primary"
        size="lg"
        class="icon-btn icon-btn-right"
        flat
        @click.prevent="openChartModal"
      />
    </div>

    <div
      v-if="displayAccounts"
      class="header-accounts-button-wrapper"
    >
      <q-btn
        icon="people"
        color="primary"
        size="lg"
        class="icon-btn icon-btn-right"
        flat
        @click.prevent="setAccountModalOpened(true)"
      />
    </div>


  </section>
</template>

<script>
import { mapState } from 'vuex';
import SelectAccount from '@/components/Modals/SelectAccount';

export default {
  name: 'Header',
  components: {
    SelectAccount,
  },
  data() {
    return {
      isBackButtonEnabled: false,
    };
  },
  computed: {
    ...mapState({
      id: state => state.route.params.id,
      authenticatedAccount: state => state.settings.authenticatedAccount,
    }),
    wallet() {
      return this.$store.getters['entities/wallet/find'](this.id);
    },
    displaySettings() {
      // if (this.authenticatedAccount) return true;
      return false;
    },
    displayAccounts() {
      return this.$route.path === '/';
    },
    displayAddWallet() {
      return this.$route.path === '/wallet';
    },
    displayPriceChart() {
      return this.$route.name === 'walletSingle';
    },
    supportedCoins() {
      return this.$store.state.settings.supportedCoins;
    },
    heading() {
      if (this.$route.name === 'setup') return '';
      if (this.$route.name === 'exchange') return 'Exchange';
      if (this.$route.name === 'settings') return 'Settings';
      return 'CryptoWallet';
    },
    coinHeading() {
      if (this.$route.name === 'walletSingle' ||
          this.$route.name === 'coinSinglePrices') {
        return true;
      }
      return false;
    },
    hideHeader() {
      if (this.$route.path === '/setup/0') return true;
      return false;
    },
    coinLogo() {
      const coin = this.supportedCoins.find(cc => cc.name === this.wallet.name);
      /* eslint-disable-next-line */
      return require(`@/assets/cc-icons/color/${coin.symbol.toLowerCase()}.svg`);
    },
  },
  watch: {
    $route() {
      if (window.history.length > 0 &&
          this.$route.path !== '/setup/0' &&
          this.$route.path !== '/' &&
          this.$route.path !== '/wallet') {
        this.isBackButtonEnabled = true;
      } else {
        this.isBackButtonEnabled = false;
      }
    },
  },
  methods: {
    goBack() {
      this.$router.go(-1);
    },
    goToSettings() {
      this.$router.push({ path: '/settings' });
    },
    setAccountModalOpened(value) {
      this.$root.$emit('selectAccountModalOpened', value);
    },
    openWalletsModal() {
      this.$root.$emit('walletsModalOpened', true);
    },
    openChartModal() {
      this.$router.push({ path: `/wallet/single/prices/${this.wallet.id}` });
    },
  },
};
</script>

<style>
.header-section {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 2.5rem;
  font-family: 'CooperHewitt-Semibold';
  background: #1e3c57;
  position: relative;
}

.header-h1 {
  letter-spacing: normal;
  font-size: 1em;
  margin-top: 0.3em;
  position: relative;
}

.header-h1.logo {
  font-family: 'CooperHewitt-BoldItalic';
}

.header-back-button-wrapper {
  position: absolute;
  left: 0.5rem;
  height: 100%;
  display: flex;
  align-items: center;
}

.header-settings-button-wrapper,
.header-accounts-button-wrapper {
  position: absolute;
  right: 0.5rem;
  height: 100%;
  display: flex;
  align-items: center;
}

.header-accounts-button-wrapper {
  right: .8rem;
}

.icon-btn {
  margin: 0;
  padding-bottom: 0;
  min-height: auto;
  height: 100%;
  position: relative;
  width: 4em;
  padding: 0;
}

.back-arrow-btn .q-btn-inner {
  justify-content: flex-start;
}

.icon-btn-right .q-btn-inner {
  justify-content: flex-end;
}

.back-arrow-btn .q-focus-helper {
  display: none;
}

.coin-logo {
  height: 1.25rem;
  margin-right: 0.2rem;
  position: relative;
  top: 0.2em;
}
</style>
