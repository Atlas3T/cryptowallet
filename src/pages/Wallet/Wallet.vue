<template>
  <div class="layout">
    <div class="list">
      <WalletsList
        :coins="coins"
        :currency="currency"
      />
    </div>
    <div class="button">
      <AddWallet/>
    </div>
  </div>
</template>

<script>
import WalletsList from '@/components/Wallet/Wallet/WalletsList.vue';
import AddWallet from '@/components/Wallet/Wallet/AddWallet.vue';

export default {
  name: 'Balance',
  components: {
    WalletsList,
    AddWallet,
  },
  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      currency: 'GPB',
      coins: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
