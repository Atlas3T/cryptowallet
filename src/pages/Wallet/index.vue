<template>
  <div class="layout">
    <div class="list">
      <wallet-list
        :wallets="wallets"
        :currency="selectedCurrency"
      />
    </div>
    <div class="button">
      <add-wallet/>
    </div>
  </div>
</template>

<script>
import WalletList from '@/components/Wallet/WalletsList';
import AddWallet from '@/components/Wallet/AddWallet';

export default {
  name: 'Balance',

  components: {
    WalletList,
    AddWallet,
  },

  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      wallets: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },

  computed: {
    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
