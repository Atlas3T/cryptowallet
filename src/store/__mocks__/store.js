import Vue from 'vue';
import Vuex from 'vuex';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from '@atlascity/vuex-orm-plugin-lokijs';

import wallet from '@/store/wallet';
import contacts from '@/store/contacts/__mocks__/contacts.js';
import search from '@/store/search/__mocks__/search.js';
import settings from '@/store/settings/__mocks__/settings.js';
import setup from '@/store/setup/__mocks__/setup.js';
import qrcode from '@/store/qrcode/__mocks__/qrcode.js';
import modals from '@/store/modals/__mocks__/modals.js';

// entities
import Account from '@/store/wallet/entities/account';
import Address from '@/store/wallet/entities/address';
import Tx from '@/store/wallet/entities/tx';
import Utxo from '@/store/wallet/entities/utxo';
import Wallet from '@/store/wallet/entities/wallet';
import Coin from '@/store/wallet/entities/coin';
import KeyPair from '@/store/wallet/entities/keyPair';
import Prices from '@/store/prices';
import LatestPrice from '@/store/latestPrice';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
  Wallet,
];

const options = {
  env: 'browser',
};

VuexORM.use(VuexORMLoki, { database, options });

export const actions = {};
export const getters = {};
export const mutations = {};
export const state = {};

export function createMocks(custom = {
  getters: {}, mutations: {}, actions: {}, state: {},
}) {
  /**
   * mock store modules.
   */
  const modules = {
    wallet,
    search,
    settings,
    contacts,
    setup,
    qrcode,
    modals,
  };

  let mockGettersCombined = {};
  let mockMutationsCombined = {};
  let mockActionsCombined = {};
  let mockStateCombined = {};

  Object.entries(modules).forEach(([name, module]) => {
    mockGettersCombined = Object.assign({}, mockGettersCombined, module.getters);
    mockMutationsCombined = Object.assign({}, mockMutationsCombined, module.mutations);
    mockActionsCombined = Object.assign({}, mockActionsCombined, module.actions);
    mockStateCombined = Object.assign({}, mockStateCombined, module.state);

    if (custom.state && custom.state[name]) {
      module.state = Object.assign(module.state, custom.state[name]);
    }

    modules[name] = {
      namespaced: true,
      getters: Object.assign({}, getters, module.getters),
      mutations: Object.assign({}, mutations, module.mutations),
      actions: Object.assign({}, actions, module.actions),
      state: Object.assign({}, state, module.state),
    };
  });

  mockGettersCombined = Object.assign({}, mockGettersCombined, custom.getters);
  mockMutationsCombined = Object.assign({}, mockMutationsCombined, custom.mutations);
  mockActionsCombined = Object.assign({}, mockActionsCombined, custom.actions);
  mockStateCombined = Object.assign({}, mockStateCombined, custom.state);
  return {
    getters: mockGettersCombined,
    mutations: mockMutationsCombined,
    actions: mockActionsCombined,
    state: mockStateCombined,
    store: new Vuex.Store({
      getters: mockGettersCombined,
      mutations: mockMutationsCombined,
      actions: mockActionsCombined,
      state: mockStateCombined,
      modules,
      plugins: [VuexORM.install(database)],
    }),
  };
}

export const { store } = createMocks();
