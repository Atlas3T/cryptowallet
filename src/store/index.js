/*eslint-disable*/

import Vue from 'vue';
import Vuex from 'vuex';
import VuexPersistence from 'vuex-persist';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from 'vuex-orm-lokijs';

// import entities.
import Account from './wallet/entities/account';
import Address from './wallet/entities/address';
import Tx from './wallet/entities/tx';
import Utxo from './wallet/entities/utxo';
import Wallet from './wallet/entities/wallet';
import Coin from './wallet/entities/coin';
import KeyPair from './wallet/entities/keyPair';
import Prices from './prices';
import LatestPrice from './latestPrice';

// import modules.
import payments from './payments';
import search from './search';
import settings from './settings';
import setup from './setup';
import contacts from './contacts';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
];

window.Tx = Tx;
window.Utxo = Utxo;
window.Address = Address;
window.Wallet = Wallet;
window.LatestPrice = LatestPrice;
window.Prices = Prices;
window.Account = Account;
window.Coin = Coin;

// Setup persistant storage.
const vuexLocal = new VuexPersistence({
  supportCircular: true,
  storage: window.localStorage,
  modules: [settings],
});


const options = {
  env: 'browser',
};

function hydrationCompletedCallback() {
  setTimeout(() => {
    store.dispatch('settings/setLoading', false);
  }, 1000);
}

VuexORM.use(VuexORMLoki, { database, options, hydrationCompletedCallback });


/**
 * Create CryptoWallet Vuex store obj.
 * @type {Store<any>}
 */
const store = new Vuex.Store({
  modules: {
    payments,
    search,
    settings,
    setup,
    contacts,
  },
  plugins: [VuexORM.install(database)],
});

Vue.prototype.activeWallets = {};

// @todo remove line below (used for development purposes only)
window.store = store;

export default store;
