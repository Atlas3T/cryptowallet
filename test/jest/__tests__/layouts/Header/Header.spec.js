/* eslint-disable no-magic-numbers */
import { mount } from '@vue/test-utils';
import Header from '@/layouts/Header';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';
import Wallet from '@/store/wallet/entities/wallet';
import LatestPrice from '@/store/latestPrice';

describe('Header.vue', () => {
  let storeMocks;
  let wrapper;
  let router;

  const walletData = JSON.parse('{"$id":3,"id":3,"account_id":1,"name":"Ethereum","displayName":"Ethereum","symbol":"ETH","sdk":"Ethereum","network":"ETHEREUM","internalChainAddressIndex":0,"externalChainAddressIndex":1,"externalAddress":null,"confirmedBalance":10,"unconfirmedBalance":0,"imported":true,"enabled":true,"lastBlockHeight":0,"parentName":"","parentSdk":"","contractAddress":"","decimals":"","hdWallet":{"bip":49,"ext":{"xpriv":"xprv9zZZrtcYB6bg2sxhoVYYKAM8duZHP1MwV1SYNVLStU6qpY6NCKfuTCeeHe1MsBeYDt7XipqZYtshoGC9ghNZ6EVwugKxUPM7m2tFmExu2mt","xpub":"xpub6DYvGQ9S1U9yFN3AuX5YgJHsBwPmnU5nrEN9Ask4SodphLRWjrz9zzy88v6HFPGJuDoMg7hhEG2yFFhU7Djqx3AoE57hEL748iFp26JLXtC"},"int":{"xpriv":"xprv9zZZrtcYB6bg6LXLr8kweVuqCWUTc5dnFDJ526bEyz5pLHdfHRx2D1UPLE8bZy6hK8f1YpXQHeGF2FWEds5S1MSbTH2boaQRNUoz6vy4MRq","xpub":"xpub6DYvGQ9S1U9yJpboxAHx1drZkYJx1YMdcSDfpUzrYKcoD5xopyGGkonsBXbmCnHteYYMcjgWRU7EyByaGf8Mav4y5BTkYmC2TD2JVPmkJxs"},"type":1,"network":{"name":"BITCOIN_TESTNET","type":"testnet","bip":1,"segwit":true,"discovery":"http://92.207.178.198:3001/api","broadcastUrl":"https://chain.so/api/v2/send_tx/BTCTEST","feeApi":"https://api.blockcypher.com/v1/btc/main","connect":{"messagePrefix":"Bitcoin Signed Message:","bech32":"tb","bip32":{"public":70617039,"private":70615956},"pubKeyHash":111,"scriptHash":196,"wif":239}}},"erc20Wallet":""}');
  const erc20WalletData = JSON.parse('{"$id":4,"id":4,"account_id":1,"name":"Catalyst","displayName":"Catalyst","symbol":"CAT","sdk":"ERC20","network":"ETHEREUM","internalChainAddressIndex":0,"externalChainAddressIndex":1,"externalAddress":null,"confirmedBalance":10,"unconfirmedBalance":0,"imported":true,"enabled":true,"lastBlockHeight":0,"parentName":"Ethereum","parentSdk":"Ethereum","contractAddress":"123","decimals":"3","hdWallet":{"bip":49,"ext":{"xpriv":"xprv9zZZrtcYB6bg2sxhoVYYKAM8duZHP1MwV1SYNVLStU6qpY6NCKfuTCeeHe1MsBeYDt7XipqZYtshoGC9ghNZ6EVwugKxUPM7m2tFmExu2mt","xpub":"xpub6DYvGQ9S1U9yFN3AuX5YgJHsBwPmnU5nrEN9Ask4SodphLRWjrz9zzy88v6HFPGJuDoMg7hhEG2yFFhU7Djqx3AoE57hEL748iFp26JLXtC"},"int":{"xpriv":"xprv9zZZrtcYB6bg6LXLr8kweVuqCWUTc5dnFDJ526bEyz5pLHdfHRx2D1UPLE8bZy6hK8f1YpXQHeGF2FWEds5S1MSbTH2boaQRNUoz6vy4MRq","xpub":"xpub6DYvGQ9S1U9yJpboxAHx1drZkYJx1YMdcSDfpUzrYKcoD5xopyGGkonsBXbmCnHteYYMcjgWRU7EyByaGf8Mav4y5BTkYmC2TD2JVPmkJxs"},"type":1,"network":{"name":"BITCOIN_TESTNET","type":"testnet","bip":1,"segwit":true,"discovery":"http://92.207.178.198:3001/api","broadcastUrl":"https://chain.so/api/v2/send_tx/BTCTEST","feeApi":"https://api.blockcypher.com/v1/btc/main","connect":{"messagePrefix":"Bitcoin Signed Message:","bech32":"tb","bip32":{"public":70617039,"private":70615956},"pubKeyHash":111,"scriptHash":196,"wif":239}}},"erc20Wallet":""}');
  const latestPriceData = JSON.parse('{"$id":"ETH_GBP","coin":"ETH","currency":"GBP","updated":1554305869988,"data":{"VOLUME24HOURTO":10147125.523436274,"PRICE":3818.12,"CHANGEPCT24HOUR":4.734003741558176}}');

  function wrapperInit(options) {
    return mount(Header, options);
  }

  function storeInit(custom, route = '/') {
    storeMocks = createStoreMocks(custom);
    Wallet.$insert({ data: [walletData, erc20WalletData] });
    LatestPrice.$insert({ data: latestPriceData });
    router = createRouter(storeMocks.store);
    router.push({ path: route });
    wrapper = wrapperInit({
      i18n, router, localVue, store: storeMocks.store,
    });
  }

  beforeEach(() => { return storeInit(); });

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('it renders a section with .header-section class', () => {
    expect(wrapper.contains('section.header-section')).toBe(true);
  });

  it('renders a h1 element with .header-h1 class and the wallet name within it', () => {
    expect(wrapper.contains('h1.header-h1')).toBe(true);
    expect(wrapper.find('h1.header-h1').html().includes('CryptoWallet')).toBe(true);
  });

  it('renders a "Go Back" button', () => {
    expect(wrapper.find('button').text()).toBe('arrow_back');
  });

  it('calls a goBack method on the "Go Back" button click', (done) => {
    router.push({ path: '/fake/' });
    wrapper.find('button').trigger('click');
    setTimeout(() => {
      expect(router.history.current.path).toBe('/');
      done();
    }, 500);
  });

  it('switches the "Go Back" button disabled state based on the window.history length', () => {
    expect(wrapper.contains('button.disabled')).toBe(true);
    router.push({ path: '/fake/' });
    expect(wrapper.contains('button.disabled')).toBe(false);
  });

  it('displays a select account button if on /', () => {
    expect(wrapper.findAll('button').at(1).text()).toBe('people');
  });

  it('displays an add wallet button if on /wallet', () => {
    router.push({ path: '/wallet' });
    expect(wrapper.findAll('button').at(1).text()).toBe('add');
  });

  it('displays a price chart button if on walletSingle or sendCoinSingle and wallet has price data', (done) => {
    storeInit({}, '/wallet/single/3');
    setTimeout(() => {
      expect(wrapper.findAll('button').at(1).text()).toBe('timeline');
      done();
    }, 0);
  });

  it('goes to price chart screen when price chart button is clicked', (done) => {
    storeInit({}, '/wallet/single/3');
    wrapper.findAll('button').at(1).trigger('click');
    setTimeout(() => {
      expect(router.history.current.path).toBe('/wallet/single/prices/3');
      done();
    }, 0);
  });

  it('calls setSelectAccountModalOpened action on account button click', (done) => {
    wrapper.findAll('button').at(1).trigger('click');
    setTimeout(() => {
      expect(storeMocks.actions.setSelectAccountModalOpened.mock.calls[0][1]).toEqual(true);
      done();
    }, 0);
  });

  it('calls setAddwalletModalOpened action on add wallet button click', (done) => {
    router.push({ path: '/wallet' });
    wrapper.findAll('button').at(1).trigger('click');
    setTimeout(() => {
      expect(storeMocks.actions.setAddWalletModalOpened.mock.calls[0][1]).toEqual(true);
      done();
    }, 0);
  });

  it('hides the header if on /setup/0', () => {
    router.push({ path: '/setup/0' });
    expect(wrapper.contains('section.header-section')).toBe(false);
  });

  it('displays a generic coin logo on header if logo is found', () => {
    storeInit({}, '/wallet/single/4');
    expect(wrapper.contains('img[src="./statics/cc-icons/color/generic.svg"]')).toBe(true);
  });

  it('changes heading title based on route name', () => {
    router.push({ path: '/setup/1' });
    expect(wrapper.find('h1.header-h1').text().includes('')).toBe(true);
    router.push({ path: '/settings' });
    expect(wrapper.find('h1.header-h1').text().includes(wrapper.vm.$t('settings'))).toBe(true);
    router.push({ path: '/exchange' });
    expect(wrapper.find('h1.header-h1').text().includes(wrapper.vm.$t('exchange'))).toBe(true);
  });
});
