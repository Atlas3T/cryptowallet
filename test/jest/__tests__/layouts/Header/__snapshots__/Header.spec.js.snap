// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue renders and matches snapshot 1`] = `
<section
  class="header-section"
>
  <div
    class="header-back-button-wrapper"
  >
    <button
      class="q-btn inline q-btn-item non-selectable icon-btn back-arrow-btn q-btn--flat q-btn--rectangle text-primary disabled hideBackButton"
      disabled="disabled"
      style="font-size: 20px;"
      tabindex="-1"
      type="button"
    >
      <div
        class="q-focus-helper"
        tabindex="-1"
      />
      <div
        class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
      >
        <i
          aria-hidden="true"
          class="material-icons q-icon"
        >
          arrow_back
        </i>
      </div>
    </button>
  </div>
   
  <div>
    <div>
      <h1
        class="header-h1 logo"
      >
        
        CryptoWallet
      
      </h1>
    </div>
  </div>
   
  <!---->
   
  <!---->
   
  <div
    class="header-accounts-button-wrapper"
  >
    <button
      class="q-btn inline q-btn-item non-selectable icon-btn icon-btn-right q-btn--flat q-btn--rectangle text-primary q-focusable q-hoverable"
      style="font-size: 20px;"
      tabindex="0"
      type="button"
    >
      <div
        class="q-focus-helper"
        tabindex="-1"
      />
      <div
        class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
      >
        <i
          aria-hidden="true"
          class="material-icons q-icon"
        >
          people
        </i>
      </div>
    </button>
  </div>
</section>
`;
